SpatialGDK/Source/SpatialGDKEditor/Public/EditorExtension/LBStrategyEditorExtension.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+class UAbstractLBStrategy;
+class FLBStrategyEditorExtensionManager;
+struct FWorkerTypeLaunchSection;
+
+class FLBStrategyEditorExtensionInterface
+{
+private:
+	friend FLBStrategyEditorExtensionManager;
+	virtual bool GetDefaultLaunchConfiguration_Virtual(const UAbstractLBStrategy* Strategy, FWorkerTypeLaunchSection& OutConfiguration, FIntPoint& OutWorldDimensions) const = 0;
+};
+
+template <typename StrategyImpl, typename Implementation>
+class FLBStrategyEditorExtensionTemplate : public FLBStrategyEditorExtensionInterface
+{
+public:
+	using ExtendedStrategy = StrategyImpl;
+private:
+
+	bool GetDefaultLaunchConfiguration_Virtual(const UAbstractLBStrategy* Strategy, FWorkerTypeLaunchSection& OutConfiguration, FIntPoint& OutWorldDimensions) const override
+	{
+		return static_cast<const Implementation*>(this)->GetDefaultLaunchConfiguration(static_cast<const StrategyImpl*>(Strategy), OutConfiguration, OutWorldDimensions);
+	}
+};
+
+class FLBStrategyEditorExtensionManager
+{
+public:
+	SPATIALGDKEDITOR_API bool GetDefaultLaunchConfiguration(const UAbstractLBStrategy* Strategy, FWorkerTypeLaunchSection& OutConfiguration, FIntPoint& OutWorldDimensions);
+
+	template <typename Extension>
+	void RegisterExtension()
+	{
+		RegisterExtension(Extension::ExtendedStrategy::StaticClass(), MakeUnique<Extension>());
```

Comment:
Nit - the rvalue ref in the private RegisterExtension function doesn't actually do anything different from passing by value as you are making this with an unnamed temporary. I'd prefer to pass by value to denote ownership is transferring.
Feel free to ignore this as this isn't a strong convention we have elsewhere. Although we will probably start adopting this in the SpatialView stuff.